Editor/Halodi/PackageRegistry/Core/ScopedRegistry.cs

```diff
@@ -14,7 +14,18 @@ public class ScopedRegistry
         public bool auth;
 
         public string token;
+        public string _auth;
+        public string email;
 
+        public NPMCredential Credential => new NPMCredential()
```

Comment:
this creates a new instance each time the property is accessed, it might be better to keep a non-serialized backing field that is initialized when the object is deserialized, or when one of the fields are updated. [Although in reality none of the fields should be public](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/fields), and only have property accessors to begin with, that would actually help when determining when the new credential should be updated.

It seems a bit redundant to have the `Credential` property when the fields are already exposed for this class.

Reply:
It has been so long that I am not sure anymore why I added this one. Probably because I thought it was easier to bundle credentials in a class/struct and pass that instead of each filed separately (like in `credentialManager.SetCredential`). I guess I only used this property for convenience to quickly construct a `NPMCredential` with a values from the `ScopedRegistry`. I agree that it could be refactored to be less redundant.